server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3283,12 +3283,7 @@ private void tryStartNextTaskAfterSnapshotUpdated(ShardId shardId, ShardSnapshot
                                 repoShardId
                             );
                         } else {
-                            startShardOperation(
-                                shardsBuilder(),
-                                updatedState.nodeId(),
-                                updatedState.generation(),
-                                entry.shardId(repoShardId)
-                            );
+                            startShardSnapshot(repoShardId, updatedState.generation());
```

Comment:
No big change here really, just extracted the code that re-computes where to run the snapshot since we don't need to do the `isQueued` check twice and used it here.